cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/resource/KafkaBrokerConfigurationDiffTest.java

```diff
@@ -0,0 +1,256 @@
+        return current;
+    }
+
+    @Test
+    public void testDefaultValue() {
+        ArrayList<ConfigEntry> ces = new ArrayList<>();
+        KafkaBrokerConfigurationDiff kcd = new KafkaBrokerConfigurationDiff(getTestingCurrentConfiguration(new ArrayList<>()), getTestingDesiredConfiguration(ces), kafkaVersion, brokerId);
+        assertThat(kcd.isDesiredPropertyDefaultValue("offset.metadata.max.bytes"), is(true));
+    }
+
+    @Test
+    public void testNonDefaultValue() {
+        ArrayList<ConfigEntry> ces = new ArrayList<>();
+        ces.add(new ConfigEntry("offset.metadata.max.bytes", "4097", false, true, false));
+        KafkaBrokerConfigurationDiff kcd = new KafkaBrokerConfigurationDiff(getTestingCurrentConfiguration(ces), getTestingDesiredConfiguration(new ArrayList<>()), kafkaVersion, brokerId);
+        assertThat(kcd.isDesiredPropertyDefaultValue("offset.metadata.max.bytes"), is(false));
+    }
+
+    @Test
+    public void testCustomPropertyAdded() {
+        ArrayList<ConfigEntry> ces = new ArrayList<>();
+        ces.add(new ConfigEntry("custom.property", "42", false, true, false));
+        KafkaBrokerConfigurationDiff kcd = new KafkaBrokerConfigurationDiff(getTestingCurrentConfiguration(new ArrayList<>()), getTestingDesiredConfiguration(ces), kafkaVersion, brokerId);
+        assertThat(kcd.getDiff().unknownConfigs(kafkaVersion).size(), is(1));
+        assertThat(kcd.cannotBeUpdatedDynamically(), is(true));
+    }
+
+    @Test
+    public void testCustomPropertyRemoved() {
+        ArrayList<ConfigEntry> ces = new ArrayList<>();
+        ces.add(new ConfigEntry("custom.property", "42", false, true, false));
+        KafkaBrokerConfigurationDiff kcd = new KafkaBrokerConfigurationDiff(getTestingCurrentConfiguration(ces), getTestingDesiredConfiguration(new ArrayList<>()), kafkaVersion, brokerId);
+        assertThat(kcd.getDiff().unknownConfigs(kafkaVersion).size(), is(1));
+        assertThat(kcd.cannotBeUpdatedDynamically(), is(true));
+    }
+
+    @Test
+    public void testCustomPropertyKept() {
+        ArrayList<ConfigEntry> ces = new ArrayList<>();
+        ces.add(new ConfigEntry("custom.property", "42", false, true, false));
+        KafkaBrokerConfigurationDiff kcd = new KafkaBrokerConfigurationDiff(getTestingCurrentConfiguration(ces), getTestingDesiredConfiguration(ces), kafkaVersion, brokerId);
+        assertThat(kcd.getDiff().unknownConfigs(kafkaVersion).size(), is(0));
+        assertThat(kcd.cannotBeUpdatedDynamically(), is(false));
+    }
+
+    @Test
+    public void testChangedPresentValue() {
+        ArrayList<ConfigEntry> ces = new ArrayList<>();
+        ces.add(new ConfigEntry("min.insync.replicas", "2", false, true, false));
+        KafkaBrokerConfigurationDiff kcd = new KafkaBrokerConfigurationDiff(getTestingCurrentConfiguration(new ArrayList<>()), getTestingDesiredConfiguration(ces), kafkaVersion, brokerId);
+        assertThat(kcd.getDiff().asOrderedProperties().asMap().size(), is(1));
```

Comment:
When the diff is non empty we should also assert on what the diff actually _is_. Just concert it to a map and `assertEquals()` against `singletonMap`.